pennylane/vqe/vqe.py

```diff
@@ -73,54 +79,57 @@ class Hamiltonian:
     :doc:`/introduction/chemistry` module can be used to generate a molecular
     Hamiltonian.
 
-    .. Warning::
-
-        Hamiltonians can be constructed using Pythonic arithmetic operations. For example:
+    Hamiltonians can alternatively be constructed using Pythonic arithmetic operations. For example:
 
-        >>> qml.PauliX(0) + 2 * qml.PauliZ(0) @ qml.PauliZ(1)
+    >>> qml.PauliX(0) + 2 * qml.PauliZ(0) @ qml.PauliZ(1)
 
-        is equivalent to the following Hamiltonian:
+    is equivalent to the following Hamiltonian:
 
-        >>> qml.Hamiltonian([1, 2], [qml.PauliX(0), qml.PauliZ(0) @ qml.PauliZ(1)])
+    >>> qml.Hamiltonian([1, 2], [qml.PauliX(0), qml.PauliZ(0) @ qml.PauliZ(1)])
 
-        When Hamiltonians are defined using arithmetic operations **inside of QNodes**, constituent observables
-        may be queued as operations/an error may be thrown. Thus, Hamiltonians must be defined either outside of QNodes,
-        or inside of QNodes using the conventional method.
+    .. Warning::
+        When Hamiltonians are defined using arithmetic operations **inside of quantum functions**, constituent observables
+        may be queued as operations/an error may be thrown. Thus, Hamiltonians must be defined either outside
+        of quantum functions, or inside of quantum functions using the conventional construction method.
```

Comment:
I am not sure if this is resolved by the deletion of the check in the tape class?

Reply:
@Lucaman99 I wonder if you would know a case where this is a problem?

Reply:
My guess is removing the check fixes this?

Reply:
I think so too. Let me remove it and see if a user complains!